### PR TITLE
Refine `GetResultSink` and use `QueryExecutor` for cop/batchCop

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: ref #6518 

Problem Summary:

### What is changed and how it works?
- move `gtests_pipeline.cpp` to `gtests_pipeline_interpreter.cpp`
- use `QueryExecutor` for cop/batchCop
- Hold the `shared_ptr<PhysicalGetResultSink>` instead of `PhysicalGetResultSink &` for `GetResultSink`
- use `std::current_exception` instead of `getCurrentExceptionMessage` to catch exception in `QueryExecutor` and pipeline model

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Coprocessor/DAGDriver.cpp [hunk 1]
```diff
@@ -181,24 +181,13 @@ try
...
-            p_stream->getProfileInfo().bytes);
-
-        if constexpr (!batch)
-        {
-            // Under some test cases, there may be dag response whose size is bigger than INT_MAX, and GRPC can not limit it.
-            // Throw exception to prevent receiver from getting wrong response.
-            if (accurate::greaterOp(p_stream->getProfileInfo().bytes, std::numeric_limits<int>::max()))
-                throw TiFlashException("DAG response is too big, please check config about region size or region merge scheduler",
-                                       Errors::Coprocessor::Internal);
-        }
```
Comment (SeaRise) [inline dbms/src/Flash/Coprocessor/DAGDriver.cpp:200]: has moved to `UnaryDAGResponseWriter`

File: dbms/src/Flash/tests/gtest_pipeline_interpreter.cpp [hunk 2]
```diff
@@ -42,9 +42,30 @@ class PipelineInterpreterExecuteTest : public DB::tests::InterpreterTestUtils
         context.addExchangeReceiver("sender_1", {{"s1", TiDB::TP::TypeString}, {"s2", TiDB::TP::TypeString}, {"s3", TiDB::TP::TypeString}});
         context.addExchangeReceiver("sender_l", {{"l_a", TiDB::TP::TypeLong}, {"l_b", TiDB::TP::TypeString}, {"join_c", TiDB::TP::TypeString}});
         context.addExchangeReceiver("sender_r", {{"r_a", TiDB::TP::TypeLong}, {"r_b", TiDB::TP::TypeString}, {"join_c", TiDB::TP::TypeString}});
+        context.addExchangeReceiver("exchange", {{"s1", TiDB::TP::TypeString}, {"s2", TiDB::TP::TypeString}});
     }
 };
 
+TEST_F(PipelineInterpreterExecuteTest, SimplePipeline)
```
Comment (SeaRise) [inline dbms/src/Flash/tests/gtest_pipeline_interpreter.cpp:49]: from `gtests_pipeline.cpp`

File: dbms/src/Flash/Coprocessor/DAGContext.cpp [hunk 3]
```diff
@@ -34,6 +34,97 @@ bool strictSqlMode(UInt64 sql_mode)
 {
     return sql_mode & TiDBSQLMode::STRICT_ALL_TABLES || sql_mode & TiDBSQLMode::STRICT_TRANS_TABLES;
 }
+// for non-mpp(cop/batchCop)
```
Comment (SeaRise) [inline dbms/src/Flash/Coprocessor/DAGContext.cpp:37]:
```suggestion

// for non-mpp(cop/batchCop)
```

**Review by ywqzzy — APPROVED** [review]

File: dbms/src/Flash/Executor/DataStreamExecutor.cpp [hunk 4]
```diff
@@ -110,4 +111,27 @@ RU DataStreamExecutor::collectRequestUnit()
...
+    BaseRuntimeStatistics runtime_statistics;
+    auto & dag_context = dagContext();
+    auto root_executor_id = dag_context.getRootExecutorId();
+    const auto & profile_streams_map = dag_context.getProfileStreamsMap();
+    auto it = profile_streams_map.find(root_executor_id);
+    if (it != profile_streams_map.end())
+    {
+        for (const auto & input_stream : it->second)
+        {
+            if (auto * p_stream = dynamic_cast<IProfilingBlockInputStream *>(input_stream.get()); p_stream)
```
Comment (windtalker) [inline dbms/src/Flash/Executor/DataStreamExecutor.cpp:131]: Will the build time not collected if the root executor is `join`?

Comment (SeaRise) [reply dbms/src/Flash/Executor/DataStreamExecutor.cpp:131]: The build time will not be collected, but now the `join` must not be the root executor.

Comment (windtalker) [reply dbms/src/Flash/Executor/DataStreamExecutor.cpp:131]: Why `join` must not be the root executor? I think the root executor is only decided by the planner in TiDB? And even if `join` is not the root executor, the executors after join will not collection the build time?

Comment (SeaRise) [reply dbms/src/Flash/Executor/DataStreamExecutor.cpp]: - For MPP, the root executor must be the exchange sender.
- For cop/batch cop, the root executor must not be join.

But you are right, the build time will be lost even if the root executor isn't join.
Update to collect the execution time of `DataStreamExecutor::data_stream` directly

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**